accepted/2023/wasm-browser-threads.md

```diff
@@ -0,0 +1,268 @@
+# Multi-threading on a browser
+
+## Goals
+ - CPU intensive workloads on dotnet thread pool
+ - enable blocking .Wait APIs from C# user code on all threads
+     - Current public API throws PNSE for it
+     - This is core part on MT value proposition.
+     - If people want to use existing MT code-bases, most of the time, the code is full of locks. 
+     - People want to use existing desktop/server multi-threaded code as is.
+ - allow HTTP and WS C# APIs to be used from any thread
+     - Underlying JS object have thread affinity
+ - JSImport/JSExport interop in maximum possible extent
+ - don't change/break single threaded build. †
+
+## Lower priority goals
+ - try to make it debugging friendly
+ - implement crypto via `subtle` browser API
```

Comment:
we should expand on how this connects, i.e. why can't we use the API right now, how will this change allow us to

Reply:
I added section for it and also for lazy DLLImport